punishments.js

```diff
@@ -61,22 +69,22 @@ Punishments.loadPunishments = function () {
 				continue;
 			}
 			for (let j = 0; j < keys.length; j++) {
-				let key = keys[i];
+				let key = keys[j];
 				if (key.includes('.')) {
-					Punishments.ips.set(key, punishment);
+					ipMap.set(key, punishment);
 				} else {
-					Punishments.userids.set(key, punishment);
+					useridMap.set(key, punishment);
 				}
 			}
 		}
 	});
-};
+}
 
-Punishments.savePunishments = function () {
+function punishmentSaver(filename, legend, useridMap, ipMap) {
```

Comment:
This definitely doesn't need to work like this. I don't think there's enough duplication here for it to be one function.